### PR TITLE
FOGL-2110: Add entry for filters in asset_tracker table

### PR DESCRIPTION
**Pull request by AmandeepArora**
Tested with filter pipeline containing Asset filter and Metadata filter (in that order) on a benchmark plugin instance generating 7 assets Random1 thru' Random7.

Asset filter config rules:

curl -X PUT -d '{"value" : {"rules":[
{"asset_name":"Random1","action":"include"},{"asset_name":"Random2","action":"rename","new_asset_name":"Random82"},{"asset_name":"Random3","action":"exclude"},{"asset_name":"Random4","action":"rename","new_asset_name":"Random84"},{"asset_name":"Random5","action":"exclude"},{"asset_name":"Random6","action":"rename","new_asset_name":"Random86"},{"asset_name":"Random7","action":"include"}], "defaultAction" : "include" }
}' http://127.0.0.1:8081/foglamp/category/bench1_AssetFilter/config

Entries inserted into asset_tracker table for this setup is in following order:

48|Random1|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:19.697
49|Random2|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:19.725
50|Random82|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:19.750
51|Random3|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:19.779
52|Random4|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:19.808
53|Random84|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:19.838
54|Random1|Filter|bench1|FogLAMP|MetadataFilter|2019-01-07 16:30:19.879
55|Random82|Filter|bench1|FogLAMP|MetadataFilter|2019-01-07 16:30:19.904
56|Random84|Filter|bench1|FogLAMP|MetadataFilter|2019-01-07 16:30:19.931
57|Random1|Ingest|bench1|FogLAMP|Benchmark|2019-01-07 16:30:19.966
58|Random82|Ingest|bench1|FogLAMP|Benchmark|2019-01-07 16:30:19.990
59|Random84|Ingest|bench1|FogLAMP|Benchmark|2019-01-07 16:30:20.017
60|Random5|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:25.073
61|Random6|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:25.103
62|Random86|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:25.127
63|Random7|Filter|bench1|FogLAMP|AssetFilter|2019-01-07 16:30:25.149
64|Random86|Filter|bench1|FogLAMP|MetadataFilter|2019-01-07 16:30:25.170
65|Random7|Filter|bench1|FogLAMP|MetadataFilter|2019-01-07 16:30:25.192
66|Random86|Ingest|bench1|FogLAMP|Benchmark|2019-01-07 16:30:25.213
67|Random7|Ingest|bench1|FogLAMP|Benchmark|2019-01-07 16:30:25.235

where asset_tracker DB table schema is defined as:

CREATE TABLE asset_tracker (
       id            integer          PRIMARY KEY AUTOINCREMENT,
       asset         character(50)    NOT NULL,
       event         character varying(50) NOT NULL,
       service       character varying(255) NOT NULL,
       foglamp       character varying(50) NOT NULL,
       plugin        character varying(50) NOT NULL,
       ts            DATETIME DEFAULT (STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW', 'localtime')) );

This branch was created out of FOGL-2134 containing filtering pipeline dynamic reconfig changes.

### CODE REVIEW

File: C/common/asset_tracking.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+#include <logger.h>
+#include <asset_tracking.h>
+
+using namespace std;
+
+
+AssetTracker *AssetTracker::instance = 0;
+
+AssetTracker *AssetTracker::getAssetTracker()
```
Comment (MarkRiddoch) [inline C/common/asset_tracking.cpp]: We should have doxygen style comments on every method

File: C/common/asset_tracking.cpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ * @param plugin  	Plugin name
+ * @param event  	Event name
+ */
+void AssetTracker::populateAssetTrackingCache(string /*plugin*/, string /*event*/)
+{
+	try {
+		std::vector<AssetTrackingTuple*>& vec = m_mgtClient->getAssetTrackingTuples(m_service);
+		for (AssetTrackingTuple* & rec : vec)
+		{
+			/*if (rec->m_pluginName != plugin || rec->m_eventName != event)
```
Comment (MarkRiddoch) [inline C/common/asset_tracking.cpp]: Why is this commented out? If it is not needed then please remove the code.

File: C/common/include/asset_tracking.h [hunk 3]
```diff
@@ -74,4 +75,24 @@ namespace std
     };
 }
 
+class ManagementClient;
+
+class AssetTracker {
```
Comment (MarkRiddoch) [inline C/common/include/asset_tracking.h]: A doxygen style comment on the class should be part of our standard. I would like to start building doxygen documentation of the C++ classes

File: C/common/asset_tracking.cpp [hunk 4]
```diff
@@ -1,123 +1,123 @@
...
-	}
-	
-	AssetTrackingTuple tuple(m_service, plugin, asset, event);
-	addAssetTrackingTuple(tuple);
-}
-
+/*
+ * FogLAMP asset tracking related
+ *
+ * Copyright (c) 2019 Dianomic, LLC
```
[the remark this answers is not in the document]
Comment (AmandeepArora) [reply C/common/asset_tracking.cpp]: Updated

File: C/common/asset_tracking.cpp (hunk 2 above)
Comment (AmandeepArora) [reply C/common/asset_tracking.cpp]: Removed the code for now.

File: C/common/asset_tracking.cpp (hunk 1 above)
Comment (AmandeepArora) [reply C/common/asset_tracking.cpp]: Done.

**Comment by AmandeepArora** [comment]
@MarkRiddoch, @pintomax  - Handled review feedback.

**Pull request merged**